src/server/.tests/routes/controllers/utilities/passportLoginCllbacks.spec.js

```diff
@@ -0,0 +1,156 @@
+
+			//then
+			const responseStatus = suite.responseMock._getStatusCode()
+			const responseBody = suite.responseMock._getData()
+
+			const expectedResponseStatus = statusCodes.BAD_REQUEST
+			const expectedResponseBody = JSON.stringify({
+				description: responseMessages.errors.BAD_REQUEST
+			})
+
+			assert.strictEqual(responseBody, expectedResponseBody)
+			assert.equal(responseStatus, expectedResponseStatus)
+			done()
+		})
+
+		it('should respond with INTERNAL SERVER error when unknown error was given', done => {
+			
+			//given
+			const DUMMY_ERROR = 'Morty is smarter than Rick'
+			const DUMMY_USER = null
+			const DUMMY_INFO = null
+
+			//when
+			suite.authenticateCallback(DUMMY_ERROR, DUMMY_USER, DUMMY_INFO)
+
+			//then
+			const responseStatus = suite.responseMock._getStatusCode()
+			const responseBody = suite.responseMock._getData()
+
+			const expectedResponseStatus = statusCodes.INTERNAL_SERVER
+			const expectedResponseBody = JSON.stringify({
+				description: responseMessages.errors.INTERNAL_SERVER
+			})
+
+			assert.strictEqual(responseBody, expectedResponseBody)
+			assert.equal(responseStatus, expectedResponseStatus)
+			done()
+		})
+
+
+		it('should respond with UNAUTHORIZED ACCESS when uncompleted or invalid params was given', done => {
+			
+			//given
+			const DUMMY_ERROR = responseMessages.errors.UNAUTHORIZED
+			const DUMMY_USER = null
+			const DUMMY_INFO = null
+
+			//when
+			suite.authenticateCallback(DUMMY_ERROR, DUMMY_USER, DUMMY_INFO)
+
+			//then
+			const responseStatus = suite.responseMock._getStatusCode()
+			const responseBody = suite.responseMock._getData()
+
+			const expectedResponseStatus = statusCodes.UNAUTHORIZED
+			const expectedResponseBody = JSON.stringify({
+				description: responseMessages.errors.UNAUTHORIZED
+			})
+
+			assert.strictEqual(responseBody, expectedResponseBody)
+			assert.equal(responseStatus, expectedResponseStatus)
+			done()
+		})
+
+	})
+
+
+	describe('logIn Callback', () => {
+		beforeEach(() => {
+			suite.logInCallback = passportLoginCallbacks.logInCallback(suite.responseMock)
+		})
+
+
+		it('should respond with internal server error when error was given', done => {
+			
+			//given
+			const DUMMY_ERROR = 'Morty is smarter than Rick'
+
+			//when
+			suite.logInCallback(DUMMY_ERROR)
+
+			//then
+			
+			const responseStatus = suite.responseMock._getStatusCode()
+			const responseBody = suite.responseMock._getData()
+
+			const expectedResponseStatus = statusCodes.INTERNAL_SERVER
+			const expectedResponseBody = JSON.stringify({
+				description: responseMessages.errors.INTERNAL_SERVER
+			})
+
+			assert.strictEqual(responseBody, expectedResponseBody)
+			assert.equal(responseStatus, expectedResponseStatus)
+			done()
+		
+		})
+
+
+		it('should respond with user successfully login message when valid data was given', done => {
+
+			//given
+			const DUMMY_ERROR = null
+
+			//when
+			suite.logInCallback(DUMMY_ERROR)
```

Comment:
Is DUMMY ERROR, a valid data that leads to logging in?

Reply:
ok